pkg/controller/knativeserving/openshift/openshift.go

```diff
@@ -167,31 +178,141 @@ func checkVersion(instance *servingv1alpha1.KnativeServing) error {
+func createIngressNamespace(operatorNamespace string) string {
+	ns := &v1.Namespace{}
+	if err := api.Get(context.TODO(), client.ObjectKey{Name: operatorNamespace + "-ingress"}, ns); err != nil {
+		if apierrors.IsNotFound(err) {
+			ns.Name = operatorNamespace + "-ingress"
+			if err = api.Create(context.TODO(), ns); err != nil {
+				return ""
+			}
+			return ns.Name
+		}
+		return ""
+	}
+	return ns.Name
+}
+
+func applyServiceMesh(instance *servingv1alpha1.KnativeServing) error {
+	log.Info("Installing serviceMeshControlPlane")
+	if err := installServiceMeshControlPlane(instance); err != nil {
+		return err
+	}
+	log.Info("Successfully installed serviceMeshControlPlane")
+	log.Info("Wait ServiceMeshControlPlane condition to be ready")
+	// wait for serviceMeshControlPlane condition to be ready before reconciling knative serving component
+	if err := waitForServiceMeshControlPlaneReady(instance.GetNamespace()); err != nil {
+		return err
+	}
+	log.Info("ServiceMeshControlPlane is ready")
+	log.Info("Installing ServiceMeshMemberRole")
+	if err := installServiceMeshMemberRole(instance.GetNamespace()); err != nil {
+		return err
+	}
+	log.Info(fmt.Sprintf("Successfully installed ServiceMeshMemberRole and configured %s namespace", instance.GetNamespace()))
+	log.Info(fmt.Sprintf("Wait ServiceMeshMemberRole to update %s namespace into configured members", instance.GetNamespace()))
+	if err := waitForServiceMeshMemberRoleReady(instance.GetNamespace()); err != nil {
+		return err
+	}
+	log.Info(fmt.Sprintf("Successfully configured %s namespace into configured members", instance.GetNamespace()))
+	return nil
+}
+
+// waitForServiceMeshControlPlaneReady checks whether serviceMeshControlPlane installs all required component
+func waitForServiceMeshControlPlaneReady(operatorNamespace string) error {
+	smcp := &maistrav1.ServiceMeshControlPlane{}
+	if waitError := wait.PollImmediate(pollInterval, pollTimeout, func() (bool, error) {
+		if err := api.Get(context.TODO(), client.ObjectKey{Namespace: createIngressNamespace(operatorNamespace), Name: smcpName}, smcp); err != nil {
+			if apierrors.IsNotFound(err) {
+				return false, nil
+			}
+			return false, err
+		}
+		for _, cond := range smcp.Status.Conditions {
+			if cond.Type == maistrav1.ConditionTypeReady && cond.Status == maistrav1.ConditionStatusTrue {
+				return true, nil
+			}
+		}
+		return false, nil
+	}); waitError != nil {
+		return waitError
+	}
+	return nil
+}
+
+// installServiceMeshControlPlane installs serviceMeshControlPlane
+func installServiceMeshControlPlane(instance *servingv1alpha1.KnativeServing) error {
+	const (
+		path = "deploy/resources/serviceMesh/smcp.yaml"
+	)
+	manifest, err := mf.NewManifest(path, false, api)
+	if err != nil {
+		log.Error(err, "Unable to create serviceMeshControlPlane install manifest")
+		return err
+	}
+	transforms := []mf.Transformer{mf.InjectOwner(instance)}
+	if len(createIngressNamespace(instance.GetNamespace())) > 0 {
```

Comment:
We really shouldn't call this `create` function multiple times. We need an extra function that goes like this:

```
func ingressNamespace(ns string) string {
    return ns + "-ingress"
}
```

Reply:
`createIngressNamespace` also do samething it will create only when ns don't exist
Instead of adding extra function i will rename `createIngressNamespace` -> `ingressNamespace`
Is that okey?